### PR TITLE
Move sdk code for hybrid /react native app to their own libs

### PR DESCRIPTION
**Pull request by wmathurin**
Added two new libs: SalesforceHybrid, SalesforceReact 
Both depend on SmartSync (which depends on SmartStore which depends on SalesforceSDK). 
Hybrid apps depend on SalesforceHybrid (which depends on SmartSync and Cordova) 
React native apps depend on require SalesforceReact (which depends on SmartSync and ReactNative)

SalesforceHybrid packages are under com.salesforce.androidsdk.phonegap

SalesforceReact packages are under com.salesforce.androidsdk.reactnative

Renamed UpgradeManager to SalesforceSDKUpgradeManager
Renamed UpgradeMangerWithSmartStore to SmartStoreUpgradeManager
Upgrade managers now have an update() methods which call super.update()

Move hybrid unit tests that were in SalesforceSDKTest to ForcePluginsTest

forcedroid.js was updated so native and react native app generation should work fine

### CODE REVIEW

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import com.salesforce.androidsdk.smartsync.manager.CacheManager;
+import com.salesforce.androidsdk.smartsync.manager.MetadataManager;
+import com.salesforce.androidsdk.smartsync.manager.SyncManager;
+import com.salesforce.androidsdk.ui.LoginActivity;
+import com.salesforce.androidsdk.phonegap.ui.SalesforceDroidGapActivity;
+import com.salesforce.androidsdk.util.EventsObservable;
+import com.salesforce.androidsdk.util.EventsObservable.EventType;
+
+/**
+ * SDK Manager for all applications that use SalesforceSDK and ReactNative
```
Comment (bhariharan) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: Typo: ```ReactNative``` should be replaced by ```Cordova```.

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    protected void cleanUp(Activity frontActivity, Account account) {
+    	final UserAccount userAccount = SmartSyncUserAccountManager.getInstance().buildUserAccount(account);
+    	MetadataManager.reset(userAccount);
+
+    	/*
+    	 * We don't have to do a hard reset on the cache manager here, since
+    	 * the underlying database will be wiped in the super class.
+    	 */
+    	CacheManager.softReset(userAccount);
```
Comment (bhariharan) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: Doesn't the ```super()``` call reset all these managers in ```SmartSync``` anyway?

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    	SyncManager.reset();
+        super.cleanUp(frontActivity, account);
+    }
+
+    @Override
+    public UserAccountManager getUserAccountManager() {
+    	return SmartSyncUserAccountManager.getInstance();
```
Comment (bhariharan) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: Won't this come from the ```super()``` call too?

File: settings.gradle [hunk 4]
```diff
@@ -2,6 +2,9 @@ include 'external:cordova:framework'
 include 'libs:SalesforceSDK'
 include 'libs:SmartStore'
 include 'libs:SmartSync'
+include 'libs:SmartSync'
```
Comment (bhariharan) [inline settings.gradle]: Duplicate entry of ```'libs:SmartSync'```.

File: tools/sdk.sh [hunk 5]
```diff
@@ -60,7 +62,6 @@ usage ()
     echo "        ForcePluginsTest"
     echo "    <test_target> can be "
     echo "        all"
-    echo "        ForcePluginsTest"
```
Comment (bhariharan) [inline tools/sdk.sh]: Isn't ```ForcePluginsTest``` still a test target? Or should this be replaced by ```SalesforceHybrid```?

**Comment by bhariharan** [comment]
Looks pretty good overall. Just a couple of minor comments.

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java (hunk 2 above)
Comment (wmathurin) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: You're right. I created SalesforceHybridSDKManager.java by copying SmartSyncSDKManager.java and then cleaning up - and I missed that method.

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java (hunk 1 above)
Comment (wmathurin) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: Will do.

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java (hunk 3 above)
Comment (wmathurin) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/app/SalesforceHybridSDKManager.java]: Will fix.

File: tools/sdk.sh (hunk 5 above)
Comment (wmathurin) [inline tools/sdk.sh]: It is. But it was twice in the list (look a few lines below).

File: settings.gradle (hunk 4 above)
Comment (wmathurin) [inline settings.gradle]: Will fix.

**Comment by wmathurin** [comment]
@bhariharan fixed the issues you raised (and a few more) with https://github.com/wmathurin/SalesforceMobileSDK-Android/commit/342c6f7275b7c63cf6d1c73d0fb057ac849a1af8

**Comment by bhariharan** [comment]
Awesome! :+1: 

I'll fix ```postpublish.js``` in ```CordovaPlugin``` once this goes in.

**Comment by wmathurin** [comment]
With https://github.com/wmathurin/SalesforceMobileSDK-Android/commit/05d34e9d4a10248adadde77e78b91f12f5e40ce5
ForcePluginsTests is now SalesforceHybridTest and lives under libs/test
So only the lib SalesforceReact is without tests

**Comment by wmathurin** [comment]
Last couple of checkins: small fixes in forcedroid (one in the next steps, the other for react native apps) - so not really related to this pull request.

**Pull request merged**